File: bank-service/src/main/java/rs/edu/raf/exceptions/WrongLoanAmountException.java
package rs.edu.raf.exceptions;

import org.springframework.http.HttpStatus;

import java.math.BigDecimal;

public class WrongLoanAmountException extends CustomException{
    public WrongLoanAmountException(BigDecimal maxLoanAmount) {
        super("Loan amount can't be greater than " + maxLoanAmount, HttpStatus.BAD_REQUEST);
    }
}

File: bank-service-kotlin/src/main/kotlin/org/example/bankservicekotlin/exception/WrongLoanAmountException.kt
package org.example.bankservicekotlin.exception

import org.springframework.http.HttpStatus
import java.math.BigDecimal

data class WrongLoanAmountException(val maxLoanAmount: BigDecimal) :
    CustomException("Loan amount can't be greater than $maxLoanAmount", HttpStatus.BAD_REQUEST)
